### PR TITLE
Fix snapshot content is blank

### PR DESCRIPTION
**Pull request by emcelroy**
PT Story: https://www.pivotaltracker.com/story/show/181808855

[#181808855]

Adds `InitMessageContext` context and `useContextInitMessage` hook. `useContextInitMessage` should be used instead of `useInitMessage` in components that use `BaseApp` or `BaseQuestionApp`. This will limit calls to `useInitMessage` to just one which is desired since multiple calls can lead to unnecessary additional renders.

### CODE REVIEW

File: src/shared/hooks/use-context-init-message.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { createContext, useContext } from "react";
+import { IInitInteractive } from "@concord-consortium/lara-interactive-api";
+
+export const InitMessage = createContext<IInitInteractive | null>(null);
```
Comment (kswenson) [inline src/shared/hooks/use-context-init-message.ts:4]: I would name this `InitMessageContext` to make the distinction between the message and the context that wraps it clear, and then client code would look like:
```
      <InitMessageContext.Provider value={initMessage}>
        { renderMode() }
      </InitMessageContext.Provider>
```

Comment (emcelroy) [reply src/shared/hooks/use-context-init-message.ts:4]: @kswenson Thanks. Makes sense to me.

File: src/drawing-tool/components/take-snapshot.tsx [hunk 2]
```diff
@@ -45,7 +45,7 @@ export const TakeSnapshot: React.FC<IProps> = ({ authoredState, interactiveState
     <>
       {
         snapshotTarget &&
-        <button className={cssHelpers.apButton} onClick={handleSnapshot} disabled={snapshotInProgress} data-test="snapshot-btn">
+        <button className={cssHelpers.apButton} onClick={handleSnapshot} disabled={snapshotInProgress} data-testid="snapshot-btn">
```
Comment (scytacki) [inline src/drawing-tool/components/take-snapshot.tsx:48]: @tejal-shah is this going to break any integration tests? 
This is the snapshot button used in the image question  interactive, drawing tool interactive, and labbook interactive.

**Review by scytacki — CHANGES_REQUESTED** [review]
This empty file seems to have been commit:
src/shared/utilities/linked-interactive-id.ts

It would be nice to change `useLinkedInteractives` so it only runs when the `initMessage.mode` is `authoring`.  This would make sure that none of the runtime code is depending on the authoring state being modified.  Based on the author changes you've made I would expect this to work.

**Pull request merged**